### ISSUE TITLE
File level rules aren't used in some cases

### ISSUE THREAD

**Issue by rahatarmanahmed** [eslint/eslint#3114]
Experiencing some very bizarre behavior. I've got a file with the following:
```javascript
/* eslint-env es6:false */
/* eslint no-var:0 */
var app = require('app');
var BrowserWindow = require('browser-window');
var path = require('path');
```
The env and rule are of course enabled in my `.eslintrc`. I have an npm script that runs eslint without any errors. However, when the script is run in a [TravisCI build](https://travis-ci.org/squelch-irc/squelch/jobs/72017175), it's as if those inline rules are ignored, and an error occurs. Strangely enough, I get the same error behavior when linting with a globally installed eslint with the same version (v0.24.1).

I have no clue what could be causing this.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Can you run globally as you have been running but run it with this command
`DEBUG=eslint:* eslint `

And please share the output.

What kind of parser are you using?

**Comment by gyandeeps** [comment]
FYI, you cannot disable `es6` environment inline inside a file. 
Reference: https://github.com/eslint/eslint/issues/2614

**Comment by duereg** [comment]
I'm also seeing this issue. This only started very recently - I'm guessing an underlying dependency must have changed. 

My file and line comments are being ignored

**Comment by gyandeeps** [comment]
Are you guys `babel-eslint` as a parser?

**Comment by duereg** [comment]
yes, I'm using babel-eslint

**Comment by gyandeeps** [comment]
They released a new version like 30 mins back and they changed things with respect to comments.
Even their tests are failing.
@hzoo

**Comment by duereg** [comment]
gotcha, I'll pin the correct version. Thanks!

**Comment by gyandeeps** [comment]
I am not sure if thats the root cause. Because that what has changed recently. 
I would open an issue on their repo.

**Comment by gyandeeps** [comment]
Apparently it was a known issue: https://github.com/facebook/react/pull/4449
I think with the recent release (like 40 mins back) they have fixed it.
Give it a shot again.

**Comment by rahatarmanahmed** [comment]
Can confirm, updating to `babel-eslint@3.1.24` fixed the issue.
